### ISSUE TITLE
Webpack Manifest Plugin

### ISSUE THREAD

**Issue by pustovalov** [webpack/webpack#11562]
## Feature request

Docs about manifest for webpack  https://webpack.js.org/guides/output-management/#the-manifest says:

>The manifest data can be extracted into a json file for easy consumption using the WebpackManifestPlugin.

But 
https://github.com/danethurber/webpack-manifest-plugin
https://www.npmjs.com/package/webpack-manifest-plugin

was deprecated by the author

Most likely this plugin will stop working with webpack 5:
https://github.com/danethurber/webpack-manifest-plugin/issues/186

**What is motivation or use case for adding/changing the behavior?**
many developers use this plugin:
https://www.npmtrends.com/webpack-manifest-plugin (2.6M per week)

**How should this be implemented in your opinion?**
create `webpack-manifest-plugin` by `webpack` or `webpack-contrib` organization

**Are you willing to work on this yourself?**
maybe

**Comment by evilebottnawi** [comment]
We have `stats` and you can get information from stats, why do not use `stats`?

**Comment by rkumorek** [comment]
@pustovalov I am guessing you will still be able to use https://github.com/webdeveric/webpack-assets-manifest in place of `webpack-manifest-plugin` if you prefer to, no?

**Comment by pustovalov** [comment]
@rkumorek at the moment I'm using https://github.com/webdeveric/webpack-assets-manifest like as a third level dependency. 

But there are some problems with that package and I have been waiting for a response from the author for almost a year
https://github.com/webdeveric/webpack-assets-manifest/pull/59

>We have stats and you can get information from stats, why do not use stats?

@evilebottnawi thx for your answer, I will check if that functionality can be achieved by using `stats`

**Comment by danethurber** [comment]
@evilebottnawi lack of support/maintenance mostly. i haven't contributed since 2016 and the other people that were involved are not longer around. I have no plans to maintain the project going further

**Comment by Alexendoo** [comment]
re: `stats`, I found it extremely difficult to use it to get a mapping between output filename and input filename. If the output filename can be predicted ahead of time e.g. because it contains `[name]` then `assets` can be searched easily enough.

If the output filename is not uniquely predictable then the best I could come up with is to search `modules`, sometimes the module will have `.assets` populated with the filename and you're done. Sometimes you get a chunk id and can guess which of the `.files`/`.auxilaryFiles` in said chunk is the right one but sometimes that is also non obvious. Sometimes you get neither `.assets` nor `.chunks`. Often plugins emit modules with more complicated names, e.g. `css ./node_modules/css-loader/dist/cjs.js!./example.css`. That is about where I gave up

**Comment by evilebottnawi** [comment]
I think we can improve stats for this, but it can require more time for this, because a lot of plugins and loaders don’t return original names, anyway we need to solve this

**Comment by MarcelGerber** [comment]
> It's Back!
> This repo was recently transferred and we're going to be giving the project a refresh. Please do continue to add your issues and pull requests and we'll get to them as soon as is possible. 🍻

-- https://github.com/shellscape/webpack-manifest-plugin

**Comment by piecyk** [comment]
Yeah, would be good to have this mapping out of box. It's must have for ssr, also reduce custom code on plugin side. For example https://github.com/jantimon/html-webpack-plugin generate it for injecting assets.

**Comment by ScriptedAlchemy** [comment]
SSR chunk mappings would be helpful. Especially with module federation, it would be nice to have a simple mapping that can be accessed. Stuff like webpack flush chunks is quite intense to track down chunks from stats. 

We can also use startup code and have a remote container ship it’s chunk correlation to other SSR hosts

**Comment by piernik** [comment]
Any news on this one?

**Comment by alexander-akait** [comment]
bump
